src/k8s/pkg/k8sd/features/calico/network.go

```diff
@@ -65,10 +72,15 @@ func ApplyNetwork(ctx context.Context, snap snap.Snap, cfg types.Network, _ type
 		},
 		"installation": map[string]any{
 			"calicoNetwork": map[string]any{
-				"ipPools": podIpPools,
+				"ipPools":                    podIpPools,
+				"nodeAddressAutodetectionV4": config.autodetectionV4,
+				"nodeAddressAutodetectionV6": config.autodetectionV6,
```

Comment:
i'm worried if passing `nodeAddressAutodetectionV4: null` in the values afffects or unsets any defaults that may come from the chart, breaking deployments. can you make sure that this is handled?

Reply:
I would make the `config` carry the `autodetectionV4{Key,Value}`, then set these fields only if both are set.